bin/basket.js

```diff
@@ -14,88 +14,115 @@ var config = require('../config').getProperties()
-    return true
-  }
-  if (email.match(/@restmail.lcip.org$/)) {
-    return true
-  }
-  return false
+var messageHandlers = {
+  verified: onVerified,
+  login: onLogin
 }
 
-
 var basketQueue = new SQSReceiver(config.basket.region, [config.basket.queueUrl])
 basketQueue.on(
   'data',
   function basketRequest(message) {
     log.trace({ op: 'basketRequest', sqs: message })
-    if (message.event === 'verified') {
-      // Ignore email addresses that are clearly from dev testing.
-      if (shouldIgnoreEmail(message.email)) {
-        message.del()
-        return
-      }
-      // Forward all others to basket API.
-      request(
-        {
-          url: config.basket.apiUrl + '/fxa-register/',
-          strictSSL: true,
-          method: 'POST',
-          headers: {
-            'X-API-Key': config.basket.apiKey
-          },
-          form: {
-            fxa_id: message.uid,
-            email: message.email,
-            // Basket won't accept empty or null `accept_lang` field,
-            // so we default to en-US.  This should only happen if
-            // the user has not sent an explicit Accept-Language header.
-            accept_lang: message.locale || 'en-US'
-          }
-        },
-        function (err, res, body) {
-          // Log and retry for network-level errors.
-          if (err) {
-            log.error(
-              {
-                op: 'basketRequest',
-                uid: message.uid,
-                locale: message.locale,
-                err: err
-              }
-            )
-            return
-          }
-          // Log at error level for HTTP-level errors, info level otherwise.
-          if (res.statusCode < 200 || res.statusCode >= 300) {
-            log.error(
-              {
-                op: 'basketRequest',
-                status: res.statusCode,
-                uid: message.uid,
-                locale: message.locale,
-                body: body
-              }
-            )
-          } else {
-            log.info(
-              {
-                op: 'basketRequest',
-                status: res.statusCode,
-                locale: message.locale,
-                body: body
-              }
-            )
-          }
-          message.del()
-        }
-      )
-    }
-    else {
-      message.del()
+
+    var messageHandler = messageHandlers[message.event]
+    if (messageHandler) {
+      return messageHandler(message)
     }
+
+    message.del()
   }
 )
 basketQueue.start()
+
+function onVerified (message) {
+  forwardEvent(message, '/fxa-register/', {
+    fxa_id: message.uid,
+    email: message.email,
+    // Basket won't accept empty or null `accept_lang` field,
+    // so we default to en-US.  This should only happen if
+    // the user has not sent an explicit Accept-Language header.
+    accept_lang: message.locale || 'en-US'
+  }, function (data) {
+    data.uid = message.uid
+    data.locale = message.locale
+    return data
+  })
```

Comment:
One stylistic nit: this `decorateLog` function kinda looks like a continuation callback, it look me a little while to figure out what it was doing.  Could we get away with just logging all poperties from `message` as part of the log entry, rather than letting each function specify a custom decorator?